kanidmd_web_ui/src/login.rs

```diff
@@ -39,6 +39,7 @@ enum LoginState {
     // MechChoice
     // CredChoice
     Password(bool),
+    BackupCode(bool),
```

Comment:
As much as I appreciate you adding this, I think you said there were some issues with the webui, so do you want to revert the changes in this file, and I'll resolve them in my current work? The webui atm is only a proof of concept of login, so it's not going to be breaking for most people.

Reply:
Yes, the webui will fail if you have more than one AuthMech to choose from.

I tried to revert my changes, but cannot leave it like it is in master, as I get this:

> login.rs:406:35
>     |
> 406 | ...                   match allowed.pop().unwrap() {
>     |                             ^^^^^^^^^^^^^^^^^^^^^^ pattern `BackupCode` not covered
> 
That's why I started to modify the webui initially. 

If you are okay with it, I would say we leave it like this now. The project will compile and you can use the webui with  accounts that present only one AuthMech.
Then, you can work on the improved webui and just throw it away - no problem from my side.